apis/v1alpha2/grpcroute_types.go

```diff
@@ -322,7 +322,7 @@ type GRPCMethodMatch struct {
 	// At least one of Service and Method MUST be a non-empty string.
 	// +optional
 	// +kubebuilder:validation:MaxLength=1024
-	// +kubebuilder:validation:Pattern=`^[^\/]*$`
+	// +kubebuilder:validation:Pattern=`^(?i)[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
```

Comment:
Pulling from the spec at protobuf.com. method is an `Identifier`. Translating from the BNF, I think it is correct.

Service is a [`TypeName`](https://protobuf.com/docs/language-spec#type-references), so I think there are two minor details off with this service regex though.

- It's underscores that are allowed, not dashes.
- We allow an optional leading dot

So I think the full regex should be the following ([regex101](https://regex101.com/r/bXTWuy/1)):

`^(?i)\.?[a-z_][a-z_0-9]+(\.[a-z_][a-z_0-9]+)*$`

This matches:
- `foo.bar.Baz`
- `.foo.bar.Baz`
- `.foo.bar.Bar_Baz`
- `foo`

Reply:
Thanks @gnossen! Updated to match this, PTAL